IndexedDB/idbobjectstore_createIndex15-autoincrement.htm

```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>IDBObjectStore.createIndex() - AutoIncrement in Compound Index</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="support.js"></script>
+<script>
+  indexeddb_test(
+    function(t, db, txn) {
+      // Store 1 - No auto-increment
+      var store1 = db.createObjectStore("Store1", {keyPath: "id"});
+      store1.createIndex("CompoundKey", ["num", "id"]);
+
+      // Add data
+      store1.put({id: 1, num: 100});
+    },
+    function(t, db) {
+      var store1 = db.transaction("Store1", "readwrite").objectStore("Store1");
+
+      store1.openCursor().onsuccess = t.step_func(function(e) {
```

Comment:
Rather than using a cursor then doing a get, it seems like it would be more readable to simply do a get with the expected key and check that a result was returned (i.e. `get([100, 1])`).

What are the benefits of the two-step approach here?